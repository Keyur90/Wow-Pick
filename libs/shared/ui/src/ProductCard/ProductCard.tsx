import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import VisibilityIcon from '@material-ui/icons/Visibility';
import IconButton from '@material-ui/core/IconButton';
import { FormattedAmount } from '../FormattedAmount';
import { ImageDialog } from '../ImageDialog';

interface ProductCardPropsType {
  title: string;
  code: string;
  price: number;
  image: string;
}

const useStyles = makeStyles((theme) => ({
  root: {
    '&:last-child': {
      padding: theme.spacing(2, 2, 0.5, 2),
    },
  },
  content: {
    width: '100%',
    display: 'flex',
    alignItems: 'flex-start',
    paddingBottom: theme.spacing(2),
    marginBottom: theme.spacing(0.5),
    borderBottom: '1px solid #E0E0E0',
    '& h2': {
      fontSize: '1.25rem',
      lineHeight: '1.875rem',
    },
  },
  imageButton: {
    color: '#4A90E2',
  },
  pullRight: {
    marginLeft: 'auto',
  },
}));

const ProductCard: React.FC<ProductCardPropsType> = ({ title, code, price, image }) => {
  const classes = useStyles();
  const [isImageDialogOpen, setIsImageDialogOpen] = useState(false);

  const onToggleImageDialog = (force?: boolean) => {
    setIsImageDialogOpen(force ? force : !isImageDialogOpen);
  };

  return (
    <>
      <Card>
        <CardContent className={classes.root}>
          <div className={classes.content}>
            <Typography variant="h2">{title}</Typography>
          </div>

          <Grid container spacing={2} alignItems={'center'}>
            <Grid item>
              <Typography variant="body2">{code}</Typography>
            </Grid>
            <Grid item>
              <Typography variant="body2">
                <FormattedAmount amount={price} />
              </Typography>
            </Grid>
            <Grid item className={classes.pullRight}>
              <IconButton
                className={classes.imageButton}
                color="inherit"
                aria-label="view image"
                onClick={() => onToggleImageDialog(true)}
                size={'small'}
              >
                <VisibilityIcon />
              </IconButton>
            </Grid>
          </Grid>
        </CardContent>
      </Card>
      <ImageDialog isOpen={isImageDialogOpen} onToggle={onToggleImageDialog} imageUrl={image} />
    </>
  );
};

export { ProductCard };
