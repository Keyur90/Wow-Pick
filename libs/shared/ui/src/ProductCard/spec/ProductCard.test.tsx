import React from 'react';
import { render } from '@testing-library/react';
import { ProductCard } from '../ProductCard';

test('renders ProductCard', async () => {
  const { container } = render(
    <ProductCard
      title={"Whittaker's Block Creamy Milk 33% Cocoa 250g"}
      code={751892}
      price={6.0}
      image={'https://cdn0.woolworths.media/content/wowproductimages/medium/266869.jpg'}
    />
  );

  expect(container).toMatchSnapshot();
});
