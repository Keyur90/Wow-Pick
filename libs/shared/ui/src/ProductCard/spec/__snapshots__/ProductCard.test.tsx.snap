// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductCard 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-root-1"
    >
      <div
        class="makeStyles-content-2"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          Whittaker's Block Creamy Milk 33% Cocoa 250g
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            751892
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            <span>
              $6.00
            </span>
          </p>
        </div>
        <div
          class="MuiGrid-root makeStyles-pullRight-4 MuiGrid-item"
        >
          <button
            aria-label="view image"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-imageButton-3 MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
